### PR TITLE
External Network: add internal endpoint status

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr adds an internal endpoint to the status of gateways resrouces

### CODE REVIEW

**Review by fra98 — APPROVED** [review]

File: apis/networking/v1alpha1/gatewayserver_types.go [hunk 1]
```diff
@@ -73,6 +73,14 @@ type EndpointStatus struct {
 	Protocol *corev1.Protocol `json:"protocol,omitempty"`
 }
 
+// InternalGatewayEndpoint defines the endpoint for the internal network.
+type InternalGatewayEndpoint struct {
+	// IP is the IP address of the endpoint.
+	IP IP `json:"ip,omitempty"`
```
Comment (cheina97) [inline apis/networking/v1alpha1/gatewayserver_types.go:79]: What about using a pointer to IP?

**Review by cheina97 — APPROVED** [review]

File: apis/networking/v1alpha1/gatewayserver_types.go (hunk 1 above)
Comment (cheina97) [inline apis/networking/v1alpha1/gatewayserver_types.go:79]:
```suggestion
	IP *IP `json:"ip,omitempty"`
```

**Pull request merged**